### PR TITLE
Implemented Typescript

### PR DESCRIPTION
**Pull request by NishargShah**
1. Implemented all code into Typescript
2. Added `Next.Js` as Dependency and `prettier, @types/react and @types/react-dom` as Dev Dependency
3. Root `client.js` and `index.js` changed a little bit.
4. Added `apiVersion` in every file for `stripe`
5. You can remove babel if you want.

Solved #18

### CODE REVIEW

**Comment by NishargShah** [comment]
I try to solve one problem but I am not getting the clue why its coming

```
message: "You did not provide an API key. You need to provide your API key in the Authorization header, using Bearer auth (e.g. 'Authorization: Bearer YOUR_SECRET_KEY'). See https://stripe.com/docs/api#authentication for details, or we can help at https://support.stripe.com/."
status: 401
```

If I replace the import with typescript support library it's giving me the above error when I try to access 

```
stripe.checkout.sessions.create(req.body)
```

but if I replace import with the current `beta-9` version all will be fixed, I don't understand why??

can you please guide me where I am wrong @ynnoj

**Comment by NishargShah** [comment]
> I try to solve one problem but I am not getting the clue why its coming
> 
> ```
> message: "You did not provide an API key. You need to provide your API key in the Authorization header, using Bearer auth (e.g. 'Authorization: Bearer YOUR_SECRET_KEY'). See https://stripe.com/docs/api#authentication for details, or we can help at https://support.stripe.com/."
> status: 401
> ```
> 
> If I replace the import with typescript support library it's giving me the above error when I try to access
> 
> ```
> stripe.checkout.sessions.create(req.body)
> ```
> 
> but if I replace import with the current `beta-9` version all will be fixed, I don't understand why??
> 
> can you please guide me where I am wrong @ynnoj

Solved it

**Comment by NishargShah** [comment]
@ynnoj can you please review it and merge it

**Comment by NishargShah** [comment]
xDD, I used it in my 2 projects and started the 3rd project and I thought why is it not in TS and I checked this PR still not merged, please do it ASAP, so we get clear IDE IntelliSense.

**Comment by baptisteArno** [comment]
Are you still maintaining this repo @ynnoj?

File: babel.config.json [hunk 1]
```diff
@@ -1,4 +0,0 @@
-{
-  "plugins": ["@babel/plugin-transform-runtime"],
-  "presets": [["@babel/preset-env"]]
-}
```
Comment (baptisteArno) [inline babel.config.json:4]: I think we should keep babel to build the project. It needs to use [preset/typescript](https://babeljs.io/docs/en/babel-preset-typescript)

File: package.json [hunk 2]
```diff
@@ -7,21 +7,21 @@
   "license": "MIT",
   "main": "index.js",
   "scripts": {
-    "build": "babel src --out-dir dist",
-    "watch": "babel --watch src --out-dir dist"
```
Comment (baptisteArno) [inline package.json:11]: Keep babel

**Review by baptisteArno — CHANGES_REQUESTED** [review]

**Pull request merged**